### PR TITLE
Add latitude dependent background diffusivity/viscosity

### PR DESCRIPTION
**Pull request by apcraig**
This adds a new option to the cvmix background scheme,

`config_cvmix_background_scheme = 'LatDep'
`

The background diffusivity and viscosity are implemented like "constant", but the constant values are then scaled inside the code.  config_cvmix_background_diffusion and
config_cvmix_background_viscosity specify the baseline background values.  The latitudinal dependence is hardwired into the code as specified by MV on 7/18/22.  The scaling factor varies between 0 and 1.  It is 0.25 above 42 degrees latitude, increases to 1 at 30 degrees latitude, then decreases to 0 between 27 degrees latitude and 5 degrees latitude.

This feature is fully backwards compatible.

### CODE REVIEW

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml
Comment (milenaveneziani) [inline components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml:720]: please add 'LatDep' in the 'Valid values' line below.

File: components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F
Comment (milenaveneziani) [inline components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F:177]: @apcraig: do you feel like we still need to keep the debugging option?

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml
Comment (apcraig) [reply components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml:720]: Good catch, fixed.  Although it doesn't seem to affect the ability to set it in the user_nl file which I thought is what the "valid values" was supposed to trap.  Maybe another problem in the implementation of the scripts?

File: components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F
Comment (milenaveneziani) [reply components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F:177]: maybe it would be good to have it, for when we test this PR.

Comment (apcraig) [reply components/mpas-ocean/src/shared/mpas_ocn_vmix_cvmix.F:177]: It doesn't hurt to leave it there, but have it off by default.  We're already talking about making changes to the function, so I think it's useful to leave it in.  But if others want it out, I can take it out.  It seems to be the only way to validate the implementation.

File: components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml
Comment (milenaveneziani) [reply components/mpas-ocean/bld/namelist_files/namelist_definition_mpaso.xml:720]: From what I remember in the past, I got an error on an invalid entry at runtime, not at setup. So, I am thinking the 'Valid values' are only for documentation.
At least that's my recollection.

**Comment by milenaveneziani** [comment]
Thanks @apcraig for opening this. Changes look great to me. At some point, I would add a sentence about documenting what the background_diffusivity value means for this LatDep case, but we can leave it as it for now.
I would like to test this, reproducing one of the cases that I am running using my own sandbox, once we get all the other PR's in.

**Comment by darincomeau** [comment]
Is this one ready to merge, or still requires testing?

**Comment by milenaveneziani** [comment]
If @apcraig rebases it to master after the cosp merge, then I can do a quick test (since there are namelist changes, which were not in the branch I am using now).

**Comment by apcraig** [comment]
I have done testing on Onyx and it seems to be working fine for me.  Happy for others to test as well.  @milenaveneziani, you should be able to checkout E3SMv2-Arctic and just pull this branch into your sandbox.  There are no conflicts with the COSP merge and no reason for me to rebase.  If you really want me to rebase, I can.

**Comment by milenaveneziani** [comment]
oh, if you have tested it on Onyx, then I'm fine with not testing.

**Review by milenaveneziani — APPROVED** [review]

**Comment by apcraig** [comment]
I should have noted testing in my PR, sorry about that.  I tested on Onyx with LatDep and constant background diffusivity, verified that the constant is bit-for-bit with the baseline, turned on the latdep debugging and confirmed the function is as expected, and verified that constant and latdep do no produce the same results.  I also confirm that setting the latdep debug to false runs fine and turns off the debug output.

**Comment by milenaveneziani** [comment]
I have created (and updated) the new Instruction page for Robert: https://hilat-sfa.atlassian.net/wiki/spaces/AR/pages/1730085089/Running+E3SM-Arctic+changes+to+v2+introduced+in+August+2022. @darincomeau: if you could double check it, a second pair of eyes would not hurt.
So, with that, plus this PR merged in master, we should be good to go on onyx.

Thank you both for the super quick turn around.

**Comment by apcraig** [comment]
I had a quick look and made a minor edit.  One question, do we want latdep background diffusivity on by default?  Second question, do we want to continue to test the prior grids or are we leaving those totally behind?  Once this PR is merged, I will start testing the new grids on Onyx.

**Comment by milenaveneziani** [comment]
I think for now we are testing the effect of latdep background diff. So we are running both cases.
As for the old grid: we are supposedly still in testing mode for that too, but I expect that we will only use the new mesh in the future.

**Review by darincomeau — APPROVED** [review]
Approved based on developer testing.

**Pull request merged**
